src/streaming/multiplex/advanced.rs

```diff
@@ -304,7 +304,9 @@ impl<T> Multiplex<T> where T: Dispatch {
         while self.run {
             // TODO: Only read frames if there is available space in the frame
             // buffer
-            if let Async::Ready(frame) = try!(self.dispatch.get_mut().inner.transport().poll()) {
+            if self.frame_buf.full() {
```

Comment:
I have to admit I'm not as familiar with some of the inner workings as I would like, but this is how it works as far as I can tell. If the buffer is at capacity that necessarily means that there is data to flush. If the flush needs to block that is when the task will be parked. Once the flush is finished the task will start running again. If the flush did not need to block then the task will continue running through another loop of dispatch, read, write, flush.